### DISCUSSION TITLE
v4.0.0-beta.4

### DISCUSSION THREAD

**Discussion by joschi** [dropwizard/dropwizard#6737] [Announcements]
## What's Changed
* Update dependency com.fasterxml.jackson:jackson-bom to v2.13.4.20221012 by @renovate in https://github.com/dropwizard/dropwizard/pull/6072
* Update dependency com.fasterxml.jackson:jackson-bom to v2.13.4.20221013 by @renovate in https://github.com/dropwizard/dropwizard/pull/6087
* Update dependency Sphinx to v5.3.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6089
* Update actions/cache action to v3.0.11 by @renovate in https://github.com/dropwizard/dropwizard/pull/6088
* Update dependency mysql:mysql-connector-java to v8.0.31 by @renovate in https://github.com/dropwizard/dropwizard/pull/6114
* Update github/codeql-action digest to cc7986c by @renovate in https://github.com/dropwizard/dropwizard/pull/6103
* Update mockito monorepo to v4.8.1 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/6093
* Update actions/setup-java digest to de1bb2b by @renovate in https://github.com/dropwizard/dropwizard/pull/6102
* Update dependency lesscpy to v0.15.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6128
* Update actions/upload-artifact digest to 83fd05a by @renovate in https://github.com/dropwizard/dropwizard/pull/6125
* Update dependency org.liquibase:liquibase-core to v4.17.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6129
* Update dependency org.apache.maven.plugins:maven-shade-plugin to v3.4.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6146
* Update github/codeql-action digest to 18fe527 by @renovate in https://github.com/dropwizard/dropwizard/pull/6144
* Update dependency com.uber.nullaway:nullaway to v0.10.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/6145
* Update dependency org.hibernate.orm:hibernate-core to v6.1.5.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/6147
* Update dependency com.fasterxml.jackson:jackson-bom to v2.14.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6179
* Update dependency org.cyclonedx:cyclonedx-maven-plugin to v2.7.3 by @renovate in https://github.com/dropwizard/dropwizard/pull/6190
* Update github/codeql-action digest to c3b6fce by @renovate in https://github.com/dropwizard/dropwizard/pull/6189
* Update dependency org.apache.httpcomponents.client5:httpclient5 to v5.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6192
* Update dependency org.liquibase:liquibase-core to v4.17.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6191
* Update dependency org.apache.httpcomponents.core5:httpcore5 to v5.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6193
* Update dependency org.checkerframework:checker-qual to v3.27.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6194
* Update dependency sphinx-rtd-theme to v1.1.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6195
* Fixed Possible Json Ordering Permutations Problem in Tests by @BoyuLi4 in https://github.com/dropwizard/dropwizard/pull/6134
* Revert "Update dependency jakarta.ws.rs:jakarta.ws.rs-api to v3.1.0" by @joschi in https://github.com/dropwizard/dropwizard/pull/5612
* Use Jakarta EE 9 dependencies instead of EE 10 ones by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/6215
* Update logback.version to v1.4.5 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/6224
* Update dependency com.uber.nullaway:nullaway to v0.10.5 by @renovate in https://github.com/dropwizard/dropwizard/pull/6239
* Update github/codeql-action digest to 678fc3a by @renovate in https://github.com/dropwizard/dropwizard/pull/6238
* Update dependency io.dropwizard.metrics:metrics-bom to v4.2.13 by @renovate in https://github.com/dropwizard/dropwizard/pull/6240
* Update dependency net.bytebuddy:byte-buddy to v1.12.19 by @renovate in https://github.com/dropwizard/dropwizard/pull/6241
* Update dependency org.apache.tomcat:tomcat-jdbc to v10.1.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6242
* Update dependency org.testcontainers:testcontainers-bom to v1.17.6 by @renovate in https://github.com/dropwizard/dropwizard/pull/6243
* Update slf4j.version to v2.0.4 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/6244
* Update dependency org.apache.maven.plugins:maven-install-plugin to v3.1.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6247
* Update dependency org.apache.maven.plugins:maven-install-plugin to v3.1.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6248
* Update jmh.version to v1.36 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6249
* Update mockito monorepo to v4.9.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6250
* Update dependency com.fasterxml.jackson:jackson-bom to v2.14.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6254
* Update dependency com.github.ben-manes.caffeine:caffeine to v3.1.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6256
* Update slf4j.version to v2.0.5 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/6262
* Update github/codeql-action digest to 312e093 by @renovate in https://github.com/dropwizard/dropwizard/pull/6261
* Update dependency org.checkerframework:checker-qual to v3.28.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6294
* Update actions/setup-java digest to c3ac5dd by @renovate in https://github.com/dropwizard/dropwizard/pull/6291
* Update github/codeql-action digest to b2a92eb by @renovate in https://github.com/dropwizard/dropwizard/pull/6292
* Update dependency org.jdbi:jdbi3-bom to v3.35.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6295
* Merge Renovate PRs automatically (patch only) by @joschi in https://github.com/dropwizard/dropwizard/pull/6303
* Merge Renovate PRs automatically (patch only) by @joschi in https://github.com/dropwizard/dropwizard/pull/6304
* Merge Renovate PRs automatically (patch only) by @joschi in https://github.com/dropwizard/dropwizard/pull/6305
* Update dependency org.apache.httpcomponents.client5:httpclient5 to v5.2.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6310
* Update dependency org.liquibase:liquibase-core to v4.18.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6312
* Flush appenders before stopping by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/6314
* Update github/codeql-action digest to a669cc5 by @renovate in https://github.com/dropwizard/dropwizard/pull/6320
* Cache proxy classes by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/6267
* Update dependency org.apache.tomcat:tomcat-jdbc to v10.1.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/6323
* Update dependency org.eclipse.jetty:jetty-bom to v11.0.13 by @renovate in https://github.com/dropwizard/dropwizard/pull/6325
* Update slf4j.version to v2.0.6 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/6336
* Update actions/checkout digest to 755da8c by @renovate in https://github.com/dropwizard/dropwizard/pull/6335
* Update actions/checkout action to v3.2.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6337
* Update dependency net.bytebuddy:byte-buddy to v1.12.20 by @renovate in https://github.com/dropwizard/dropwizard/pull/6354
* Update github/codeql-action digest to 959cbb7 by @renovate in https://github.com/dropwizard/dropwizard/pull/6353
* Update mockito monorepo to v4.10.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6355
* Update actions/setup-java digest to 1df8dbe by @renovate in https://github.com/dropwizard/dropwizard/pull/6352
* Update dependency org.hibernate.orm:hibernate-core to v6.1.6.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/6311
* Add migration guide for Dropwizard 4.0.x by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/6214
* Update dependency org.apache.maven.plugins:maven-invoker-plugin to v3.4.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6359
* Update dependency io.dropwizard.metrics:metrics-bom to v4.2.14 by @renovate in https://github.com/dropwizard/dropwizard/pull/6365
* Update actions/stale action to v7 by @renovate in https://github.com/dropwizard/dropwizard/pull/6370
* Update actions/cache action to v3.2.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6374
* Update dependency io.dropwizard.metrics:metrics-bom to v4.2.15 by @renovate in https://github.com/dropwizard/dropwizard/pull/6378
* Update actions/cache action to v3.2.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6385
* Update actions/cache action to v3.2.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6401
* Adjust tests to different optional behaviour of Jersey 2.38 by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/6404
* Update scacap/action-surefire-report digest to 522da11 by @renovate in https://github.com/dropwizard/dropwizard/pull/6400
* Update mockito monorepo to v4.11.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6403
* Adjust tests to different optional behaviour of Jersey 2.38 by @joschi in https://github.com/dropwizard/dropwizard/pull/6405
* Update dependency com.uber.nullaway:nullaway to v0.10.6 by @renovate in https://github.com/dropwizard/dropwizard/pull/6415
* Update error_prone.version to v2.17.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6416
* Update dependency jp.skypencil.errorprone.slf4j:errorprone-slf4j to v0.1.17 by @renovate in https://github.com/dropwizard/dropwizard/pull/6424
* Update dependency org.jdbi:jdbi3-bom to v3.36.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6425
* Update dependency org.glassfish.jersey:jersey-bom to v3.0.9 by @renovate in https://github.com/dropwizard/dropwizard/pull/6402
* Update dependency com.uber.nullaway:nullaway to v0.10.7 by @renovate in https://github.com/dropwizard/dropwizard/pull/6453
* Update dependency net.bytebuddy:byte-buddy to v1.12.21 by @renovate in https://github.com/dropwizard/dropwizard/pull/6454
* Update dependency org.cyclonedx:cyclonedx-maven-plugin to v2.7.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/6455
* Update dependency org.assertj:assertj-core to v3.24.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6456
* Update actions/checkout digest to ac59398 by @renovate in https://github.com/dropwizard/dropwizard/pull/6464
* Update actions/upload-artifact digest to 0b7f8ab by @renovate in https://github.com/dropwizard/dropwizard/pull/6465
* Update actions/cache action to v3.2.3 by @renovate in https://github.com/dropwizard/dropwizard/pull/6468
* Update scacap/action-surefire-report digest to be8172c by @renovate in https://github.com/dropwizard/dropwizard/pull/6467
* Update actions/checkout action to v3.3.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6469
* Update dependency org.checkerframework:checker-qual to v3.29.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6470
* Update error_prone.version to v2.18.0 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6474
* Update junit5 monorepo to v5.9.2 (patch) by @renovate in https://github.com/dropwizard/dropwizard/pull/6478
* Update dependency net.bytebuddy:byte-buddy to v1.12.22 by @renovate in https://github.com/dropwizard/dropwizard/pull/6502
* Update github/codeql-action digest to 515828d by @renovate in https://github.com/dropwizard/dropwizard/pull/6501
* Update dependency org.apache.httpcomponents.core5:httpcore5 to v5.2.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6503
* Update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6504
* Update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.4.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6505
* Update dependency org.apache.tomcat:tomcat-jdbc to v10.1.5 by @renovate in https://github.com/dropwizard/dropwizard/pull/6506
* Update dependency com.uber.nullaway:nullaway to v0.10.8 by @renovate in https://github.com/dropwizard/dropwizard/pull/6525
* Update dependency org.freemarker:freemarker to v2.3.32 by @renovate in https://github.com/dropwizard/dropwizard/pull/6527
* Update dependency jp.skypencil.errorprone.slf4j:errorprone-slf4j to v0.1.18 by @renovate in https://github.com/dropwizard/dropwizard/pull/6526
* Update dependency net.java.dev.jna:jna to v5.13.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6528
* Update mockito monorepo to v5 (major) by @renovate in https://github.com/dropwizard/dropwizard/pull/6529
* Update dependency org.assertj:assertj-core to v3.24.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6539
* Update dependency org.liquibase:liquibase-core to v4.19.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6540
* Update dependency mysql:mysql-connector-java to v8.0.32 by @renovate in https://github.com/dropwizard/dropwizard/pull/6556
* Update dependency org.simplify4u.plugins:pgpverify-maven-plugin to v1.17.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6562
* Update github/codeql-action digest to a34ca99 by @renovate in https://github.com/dropwizard/dropwizard/pull/6555
* Update dependency jakarta.inject:jakarta.inject-api to v2.0.1.MR by @renovate in https://github.com/dropwizard/dropwizard/pull/6577
* Update dependency maven-wrapper to v3.1.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6585
* Update github/codeql-action digest to 436dbd9 by @renovate in https://github.com/dropwizard/dropwizard/pull/6594
* Update github/codeql-action digest to 3ebbd71 by @renovate in https://github.com/dropwizard/dropwizard/pull/6597
* Update dependency com.fasterxml.jackson:jackson-bom to v2.14.2 by @renovate in https://github.com/dropwizard/dropwizard/pull/6603
* Update actions/cache action to v3.2.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/6616
* Update dependency org.apache.maven.plugins:maven-enforcer-plugin to v3.2.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6617
* Update dependency org.apache.maven.plugins:maven-enforcer-plugin to v3.2.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6618
* Update mockito monorepo to v5.1.1 (minor) by @renovate in https://github.com/dropwizard/dropwizard/pull/6620
* Switch back to subclass mock maker in `dropwizard-servlets` by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/6621
* Update dependency org.checkerframework:checker-qual to v3.30.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6628
* Update dependency net.bytebuddy:byte-buddy to v1.12.23 by @renovate in https://github.com/dropwizard/dropwizard/pull/6641
* Update dependency maven to v3.9.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6634
* Update dependency org.jdbi:jdbi3-bom to v3.37.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6635
* Update github/codeql-action digest to 39d8d7e by @renovate in https://github.com/dropwizard/dropwizard/pull/6646
* Update dependency org.hibernate.orm:hibernate-core to v6.1.7.Final by @renovate in https://github.com/dropwizard/dropwizard/pull/6659
* Update dependency sphinx-rtd-theme to v1.2.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6660
* Timestamp field comes first in generated JSON log output by @darcy-harrison in https://github.com/dropwizard/dropwizard/pull/6642
* Update actions/setup-java digest to 3f07048 by @renovate in https://github.com/dropwizard/dropwizard/pull/6664
* 4.0.x Timestamp field comes first in generated JSON log output by @darcy-harrison in https://github.com/dropwizard/dropwizard/pull/6669
* 3.0.x Timestamp field comes first in generated JSON log output by @darcy-harrison in https://github.com/dropwizard/dropwizard/pull/6668
* Update dependency com.uber.nullaway:nullaway to v0.10.9 by @renovate in https://github.com/dropwizard/dropwizard/pull/6625
* Update actions/cache action to v3.2.5 by @renovate in https://github.com/dropwizard/dropwizard/pull/6684
* Update dependency com.github.ben-manes.caffeine:caffeine to v3.1.3 by @renovate in https://github.com/dropwizard/dropwizard/pull/6685
* Update dependency org.jdbi:jdbi3-bom to v3.37.1 by @renovate in https://github.com/dropwizard/dropwizard/pull/6686
* Update github/codeql-action digest to 17573ee by @renovate in https://github.com/dropwizard/dropwizard/pull/6683
* Update dependency net.bytebuddy:byte-buddy to v1.13.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6687
* Update dependency org.apache.maven.plugins:maven-deploy-plugin to v3.1.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6688
* Update dependency org.apache.maven.plugins:maven-deploy-plugin to v3.1.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6689
* Update dependency io.dropwizard.metrics:metrics-bom to v4.2.16 by @renovate in https://github.com/dropwizard/dropwizard/pull/6695
* Update dependency org.cyclonedx:cyclonedx-maven-plugin to v2.7.5 by @renovate in https://github.com/dropwizard/dropwizard/pull/6707
* Update dependency org.checkerframework:checker-qual to v3.31.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6716
* Update dependency org.apache.maven.plugins:maven-javadoc-plugin to v3.5.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6714
* Update dependency org.apache.maven.plugins:maven-javadoc-plugin to v3.5.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6715
* Update dependency org.apache.maven.plugins:maven-invoker-plugin to v3.5.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6713
* Update dependency net.bytebuddy:byte-buddy to v1.14.0 by @renovate in https://github.com/dropwizard/dropwizard/pull/6720
* Update dependency com.github.ben-manes.caffeine:caffeine to v3.1.4 by @renovate in https://github.com/dropwizard/dropwizard/pull/6723
* Support setting base branch in release trigger workflow by @joschi in https://github.com/dropwizard/dropwizard/pull/6724
* Bump scacap/action-surefire-report from 1.3.0 to 1.4.0 by @dependabot in https://github.com/dropwizard/dropwizard/pull/6704
* Fail build at end in Java CI by @joschi in https://github.com/dropwizard/dropwizard/pull/6721
* Update actions/cache action to v3.2.6 by @renovate in https://github.com/dropwizard/dropwizard/pull/6727
* Add configurable response code meters by @dennyac in https://github.com/dropwizard/dropwizard/pull/6692
* Introduce new Dropwizard specific conversion words by @zUniQueX in https://github.com/dropwizard/dropwizard/pull/6207
* Add configurable response code meters by @joschi in https://github.com/dropwizard/dropwizard/pull/6728
* Add configurable response code meters by @joschi in https://github.com/dropwizard/dropwizard/pull/6729
* Update dependency io.dropwizard.metrics:metrics-bom to v4.2.17 by @renovate in https://github.com/dropwizard/dropwizard/pull/6733

## New Contributors
* @BoyuLi4 made their first contribution in https://github.com/dropwizard/dropwizard/pull/6134
* @darcy-harrison made their first contribution in https://github.com/dropwizard/dropwizard/pull/6642

**Full Changelog**: https://github.com/dropwizard/dropwizard/compare/v4.0.0-beta.3...v4.0.0-beta.4

<hr /><em>This discussion was created from the release <a href='https://github.com/dropwizard/dropwizard/releases/tag/v4.0.0-beta.4'>v4.0.0-beta.4</a>.</em>

**Comment by AndrewPonyk** [comment]
Why two "branches" 3.x and 4.x are in beta state?
Looks strange at first sight.

**Reply by zUniQueX** [reply]
Since we use Java/Jakarta EE libraries in Dropwizard, we have to switch the namespaces as all other libraries using those do. So 4.x is mostly 3.x under the `jakarta` namespace. However, we will further upgrade libraries in 4.x, so the branches will not stay feature equivalent.

**Reply by danjee** [reply]
Any guesses on when a GA for version 4 will come? Thanks

**Reply by joschi** [reply]
https://github.com/dropwizard/dropwizard/discussions/6066#discussioncomment-5122734
